### PR TITLE
Allow multiple Trust Anchor values in resolve requests

### LINKED ISSUE #130: Resolver with multiple trust anchors
The section on trust chain resolving makes it clear that it is feasible that an entity can have multiple trust anchors and wants to resolve a trust chain from a subject to any / multiple anchor(s) and then choose one.

At the same time the resolve endpoint does only allow one trust anchor. I think it would make sense to allow that the `anchor` parameter may be provided multiple times.

What do others think about this?

The resolver would need to choose which trust chain to use in the end; this is also with the current state the case, since there also can be multiple chains to a single trust anchor. However, the client might have preferences which trust anchor should be used, preferably. If this would be added, we might want to make a note about this. I see different options:

1. It's entirely up to the trust anchor which chain to choose and the trust anchor does not matter
2. The trust anchor should prefer the trust anchors in the order provided and only decide based on its own rules if there are multiple chains for a single trust anchor. (This would be equivalent to having the client making separate requests for all trust anchors, stopping at the first successful one)
3. The trust anchor decides based on its own rules which trust chain to use and only when having multiple equally valid options it considers the trust anchor.

I think option 3 is too vague and does not make too much sense. Option 2 could make sense.

### PR DESCRIPTION
**Pull request by selfissued**
Fixes #130

### CODE REVIEW

**Comment by peppelinux** [comment]
I request to move the trust chain in the response as REQUIRED for consistency

It makes sense to have It by default

**Comment by selfissued** [comment]
I agree that it's more consistent to always include the `trust_chain` claim in the resolve response JWT.

**Review by peppelinux — APPROVED** [review]

**Comment by zachmann** [comment]
I want to open a discussion on whether the resolver "MAY return a successful resolve response using any one of the Trust Anchor values provided" (current text of the PR) or it SHOULD return the resolve response for the first Trust Anchor that resolves successfully. 

I would argue for the later, since this gives guidance, but would still allow the resolver to do otherwise.

**Comment by peppelinux** [comment]
"First" might depend from the ordering of the trust anchor provided within the array, "Any" might mean that it is only up to the trust evaluation decide which trust anchor to use.

Any gives more freedom to the resolver in taking its choice, while "First" sounds like it would be the requestor in deciding the ordering of the trust anchor on the basis of the priority it decides

**Review by vdzhuvinov — APPROVED** [review]

**Review by rohe — APPROVED** [review]

File: openid-federation-1_0.xml [hunk 1]
```diff
@@ -4153,10 +4153,8 @@ Content-Type: application/json
 		<vspace blankLine="1"/>
 		The <spanx style="verb">anchor</spanx> request parameter
 		MAY occur multiple times, in which case,
-		the resolver may return a successful resolve response
-		using any one of the Trust Anchor values provided
-		and the <spanx style="verb">trust_chain</spanx> claim
-		MUST be included in the resolve response JWT.
+		the resolver MAY return a successful resolve response
+		using any one of the Trust Anchor values provided.
```
[the remark this answers is not in the document]
Comment (selfissued) [reply openid-federation-1_0.xml]: Deleted "one".

**Pull request merged**